### PR TITLE
sql: remove old database privilege inheritance

### PR DESCRIPTION
**Pull request by RichardJCai**
Release note (sql change): Previously, any privileges on a database
that would be inherited (copied down) to tables/schemas if a table/schema
is created in that database.

For example,
`GRANT ALL ON DATABASE TEST TO foo`
`CREATE TABLE test.t()` would result in foo having ALL on the table.

This is no longer the case as users should rely on default privileges instead.
They can achieve the same behaviour by doing
USE test;
ALTER DEFAULT PRIVILEGES GRANT ALL ON TABLES TO foo;

### CODE REVIEW

**Review by rafiss — APPROVED** [review]
lgtm, but needs test updates

File: pkg/ccl/logictestccl/testdata/logic_test/zone [hunk 1]
```diff
@@ -32,7 +32,7 @@ vectorized: true
 ·
 • scan
   missing stats
-  table: t@secondary
+  table: t@t_pkey
```
Comment (RichardJCai) [inline pkg/ccl/logictestccl/testdata/logic_test/zone:35]: I'm super confused why this file changed on rewrite...

File: pkg/ccl/logictestccl/testdata/logic_test/multi_region_privileges [hunk 2]
```diff
@@ -232,7 +233,8 @@ ALTER DATABASE repartition_privs DROP REGION "us-east-1"
 user root
 
 statement ok
-GRANT CREATE ON repartition_privs.rbr TO testuser
+GRANT CREATE ON repartition_privs.rbr TO testuser;
+GRANT CREATE ON repartition_privs.rbr2 TO testuser
```
Comment (RichardJCai) [inline pkg/ccl/logictestccl/testdata/logic_test/multi_region_privileges:237]: Need to double check if this really makes sense, having to GRANT CREATE on all tables in the database to do `SET PRIMARY REGION`

**Comment by RichardJCai** [comment]
bors r=rafiss

**Comment by rafiss** [comment]
flakes fixed by https://github.com/cockroachdb/cockroach/pull/79406

bors r+

**Pull request merged**